frame/contracts/src/wasm/runtime.rs

```diff
@@ -2367,15 +2366,14 @@ pub mod env {
 		str_ptr: u32,
 		str_len: u32,
 	) -> Result<ReturnCode, TrapReason> {
-		ctx.charge_gas(RuntimeCosts::DebugMessage)?;
+		ctx.charge_gas(RuntimeCosts::DebugMessage(str_len))?;
```

Comment:
Let's cap the `str_len` to the maximum buffer bound as the first thing in this function.